EEJT-0002/src/main/java/com/epam/esm/giftcertficate/service/GiftCertificateService.java

```diff
@@ -0,0 +1,92 @@
+package com.epam.esm.giftcertficate.service;
+
+import com.epam.esm.exceptionhandler.exception.ItemNotFoundException;
+import com.epam.esm.giftcertficate.model.GiftCertificate;
+import com.epam.esm.giftcertficate.repository.GiftCertificateRepository;
+import com.epam.esm.tag.model.Tag;
+import com.epam.esm.tag.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class GiftCertificateService {
+
+    private final GiftCertificateRepository giftCertificateRepository;
+    private final TagService tagService;
+
+    @Autowired
+    public GiftCertificateService(GiftCertificateRepository giftCertificateRepository, TagService tagService) {
+        this.giftCertificateRepository = giftCertificateRepository;
+        this.tagService = tagService;
+    }
+
+
+    public List<GiftCertificate> getAllCertificates() {
+        List<GiftCertificate> giftCertificateList = giftCertificateRepository.getAllGiftCertificates();
+        if (!giftCertificateList.isEmpty()) {
+            return giftCertificateList;
+        }
+        throw new ItemNotFoundException("There are no gift certificates");
```

Comment:
For search, get all we can return empty list as a response

Reply:
fix it , i hope in getById it is not useful to return empty list , I return list or else return ItemNotFountEx